certificateshop-logic/src/main/java/com/epam/esm/dao/impl/jdbc/UserExtractor.java

```diff
@@ -0,0 +1,110 @@
+                    addGiftCertificateToCorrespondingOrderOfUserOrders(resultSet, users, currentId, currentOrderId);
+                } else {
+                    Order newOrder = createNewOrderFromResultSetLine(resultSet);
+                    if (resultSet.getString(ColumnNames.TABLE_USERORDER_CERTIFICATE_COLUMN_CERTIFICATEINJSON) != null) {
+                        GiftCertificate giftCertificate = certificateInJsonMapper.mapRow(resultSet, resultSet.getRow());
+                        newOrder.getCertificates().add(giftCertificate);
+                        users.stream().filter(user2 -> user2.getId() == currentId)
+                                .findAny().ifPresent(user2 -> user2.getOrders().add(newOrder));
+                    }
+                }
+            } else {
+                List<Order> allOrdersThisUser = new ArrayList<Order>();
+                User user = createNewUserFromResultSetLine(resultSet, allOrdersThisUser);
+                if (resultSet.getLong(ColumnNames.TABLE_USERORDER_COLUMN_ID) != 0) {
+                    Order newOrder = createNewOrderFromResultSetLine(resultSet);
+                    user.getOrders().add(newOrder);
+                    users.add(user);
+                    addGiftCertificateToNewOrderOfUsersOrders(resultSet, users, currentId, currentOrderId, newOrder);
+                } else {
+                    users.add(user);
+                }
+            }
+        }
+        return users;
+    }
+
+    private User createNewUserFromResultSetLine(ResultSet resultSet, List<Order> allOrdersThisUser) throws SQLException {
+        User user = new User(
+                resultSet.getLong(ColumnNames.TABLE_USER_COLUMN_ID),
+                resultSet.getString(ColumnNames.TABLE_USER_COLUMN_NICKNAME),
+                allOrdersThisUser
+        );
+        return user;
+    }
+
+    private Order createNewOrderFromResultSetLine(ResultSet resultSet) throws SQLException {
+        return new Order(
+                resultSet.getLong(ColumnNames.TABLE_USERORDER_COLUMN_ID),
+                new User(resultSet.getLong(ColumnNames.TABLE_USER_COLUMN_ID),
+                        resultSet.getString(ColumnNames.TABLE_USER_COLUMN_NICKNAME),
+                        new ArrayList<>()),
+                resultSet.getTimestamp(ColumnNames.TABLE_USERORDER_COLUMN_CREATE_DATE).toLocalDateTime(),
+                resultSet.getString(ColumnNames.TABLE_USERORDER_COLUMN_NAME),
+                new ArrayList<>()
+        );
+    }
+
+    private void addGiftCertificateToCorrespondingOrderOfUserOrders(ResultSet resultSet, List<User> users, long currentId, long currentOrderId) throws SQLException {
+        if (resultSet.getString(ColumnNames.TABLE_USERORDER_CERTIFICATE_COLUMN_CERTIFICATEINJSON) != null) {
+            GiftCertificate giftCertificate = certificateInJsonMapper.mapRow(resultSet, resultSet.getRow());
+            users.stream().filter(user2 -> user2.getId() == currentId)
```

Comment:
Why do you use exactly `user2` name? Where is the first user?

Reply:
user1 is used in the line 44